compiler/optimizations/gpuTransforms.cpp

```diff
@@ -484,6 +543,13 @@ bool GpuizableLoop::callsInBodyAreGpuizableHelp(BlockStmt* blk,
   std::vector<CallExpr*> calls;
   collectCallExprs(blk, calls);
 
+  // While here, note all the GPU assertions to clean up.
```

Comment:
A comment that reads `While here, do X`, in a function that doesn't really seem related to'X' kind of screams to me: "we're not separating concerns properly".

This also seems kind of precarious (if for whatever reason this gets embedded in code that's not within a loop processed with gpuTransforms it won't get picked up). Also, I kind of wonder how this works with calls that are reachable from multiple loops:

```
foreach { foo(); }
foreach { foo(); }
proc foo() { assertOnGpu(); }
```

Now when processing the second loop that `assertOnGpu` will get processed twice.  Will `call->get(1)->remove();` assert in this case?

I think I'd do the cleanup in `cg-expr.cpp` when we process the primitives.

Reply:
Wow, that's much nicer. I was trying to avoid adding a `gCallExpr` iteration to find all `assert on gpu` calls. This does it. Thanks!